app/policies/patient_policy.rb

```diff
@@ -1,25 +1,28 @@
 class PatientPolicy < ApplicationPolicy
   def index?
-    true
+    user && record
+    (user.superuser? || user.medical_officer? || user.nurse?)
   end
 
-  def show?
-    true
-  end
-
-  def new?
-    true
-  end
-
-  alias create? new?
+  alias show? index?
+  alias new? index?
+  alias create? index?
+  alias edit? index?
+  alias update? index?
+  alias show_detail? index?
 
   class Scope < Scope
     def resolve
       return Patient.all if (user.superuser?)
 
       # CHC -> All PHCs
-      all_facilities = Facility.where(id: user.facility.id).or(Facility.where(parent_id: user.facility.id))
-      return Patient.joins(:facility).where(facilities: all_facilities) if user.secondary_nurse? || user.medical_officer?
+      all_facilities =
+        Facility
+          .where(id: user.facility.id)
+          .or(Facility.where(parent_id: user.facility.id))
+      if user.secondary_nurse? || user.medical_officer?
+        return Patient.joins(:facility).where(facilities: all_facilities)
```

Comment:
Revert this back to the ruby way.
```suggestion
return Patient.joins(:facility).where(facilities: all_facilities) if user.secondary_nurse? || user.medical_officer?
```